lib/fizzy/capi.cpp

```diff
@@ -7,10 +7,45 @@
 #include "instantiate.hpp"
 #include "parser.hpp"
 #include <fizzy/fizzy.h>
+#include <cstring>
 #include <memory>
 
 namespace
 {
+inline void set_success(FizzyError* error) noexcept
+{
+    if (error != nullptr)
+    {
+        error->code = FIZZY_SUCCESS;
+        error->message[0] = '\0';
+    }
+}
+
+inline size_t strlcpy_s(char* dest, const char* src, size_t size) noexcept
```

Comment:
Can we just make this a template over the fixed-size array so that we can have static_asserts for lengths? Not feeling safe about `assert(size > 1)` and `size - 1`.